polkadot/node/core/backing/src/lib.rs

```diff
@@ -1029,7 +1030,7 @@ fn core_index_from_statement(
 			?group_rotation_info,
 			?statement,
 			?validator_to_group,
-			n_cores = ?cores.len() ,
+			n_cores,
```

Comment:
Not for this PR, but once we got rid of the legacy stuff, I think we should make e.g. [this iterator](https://github.com/paritytech/polkadot-sdk/blob/7050c2ec1a62901dfdad80f72a41149f82d85250/polkadot/runtime/parachains/src/scheduler.rs#L200) accessible. Should enable us to be able to get rid of this parameter as well (and the fetching of the cores to begin with).